ftw/usermigration/group_members.py

```diff
@@ -0,0 +1,47 @@
+from Products.CMFCore.utils import getToolByName
+from Products.PluggableAuthService.interfaces.plugins import IGroupEnumerationPlugin
+from Products.PluggableAuthService.plugins.ZODBGroupManager import IZODBGroupManager
+
+
+def migrate_group_members(context, mapping, mode='move', replace=False):
+    """Migrate Plone users within a group."""
+
+    # Statistics
+    moved = []
+    copied = []
+    deleted = []
+
+    uf = getToolByName(context, 'acl_users')
+
+    for old_userid, new_userid in mapping.items():
+        for plugin_id, plugin in uf.plugins.listPlugins(IGroupEnumerationPlugin):
+            # Only ZODB Group Manager is supported
+            if not IZODBGroupManager.providedBy(plugin):
+                continue
+
+            for group in plugin.enumerateGroups():
+                group_id = group.get('id')
+
+                # Do nothing if a user with new_userid already exists and
+                # replace is False.
+                if new_userid in plugin.getGroupMembers(group_id) and not replace:
```

Comment:
`getGroupMembers()` is only implemented by `P.PlonePAS.interfaces.group.IGroupIntrospection` plugins. Thus you should check for that too on L19.